lib/common/client/middleware/response/snakecase.rb

```diff
@@ -24,9 +24,18 @@ def underscore_symbolize(hash)
             hash.deep_transform_keys { |k| k.underscore.to_sym }
           end
         end
+
+        class SnakecaseString < Snakecase
+          private
+
+          def underscore_symbolize(hash)
+            hash.deep_transform_keys(&:underscore)
```

Comment:
It's a lot of churn to convert to symbols. When I tried to use symbols basically everything broke in hard to diagnose ways, since a lot of these values are considered optional, so a nil is not necessarily a bad value.  Plus, once we get the values back from ActiveRecord caching, they flip back to strings, so we'd need to have a way of handling that too.